forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/RequestStateListener.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import static org.camunda.bpm.extension.commons.utils.VariableConstants.*;
+
+@Component
+public class RequestStateListener extends BaseListener implements ExecutionListener, TaskListener {
+
+
+    @Autowired
+    private HTTPServiceInvoker httpServiceInvoker;
+
+    @Override
+    public void notify(DelegateExecution execution) {
+        try {
+            invokeRequestService(execution);
+        } catch (IOException e) {
+            handleException(execution, ExceptionSource.EXECUTION, e);
+        }
+    }
+
+    @Override
+    public void notify(DelegateTask delegateTask) {
+        try {
+            invokeRequestService(delegateTask.getExecution());
+        } catch (IOException e) {
+            handleException(delegateTask.getExecution(), ExceptionSource.TASK, e);
+        }
+    }
+    
+    private void invokeRequestService(DelegateExecution execution) throws IOException {
+        ResponseEntity<String> response = httpServiceInvoker.execute(getRequestUrl(execution), HttpMethod.PUT, prepareRequestDataUrl(execution));
+        if (response.getStatusCodeValue() != HttpStatus.OK.value()) {
+            throw new ApplicationServiceException("Unable to update application " + ". Message Body: " +
+                    response.getBody());
+        }
+
+        ResponseEntity<String> auditResponse = httpServiceInvoker.execute(getRequestAuditUrl(execution), HttpMethod.POST, prepareRequestDataUrl(execution));
+        if (response.getStatusCodeValue() != HttpStatus.CREATED.value()) {
+            throw new ApplicationServiceException("Unable to capture audit for application " + ". Message Body: " +
+                    response.getBody());
+        }
+    }
+
+
+    /**
+     * Returns the endpoint of application API.
+     *
+     * @param execution
+     * @return
+     */
+    private String getRequestUrl(DelegateExecution execution) {
+        return httpServiceInvoker.getProperties().getProperty("api.url") + "/application/" + execution.getVariable(APPLICATION_ID);
+    }
+
+    protected RequestStateData prepareRequestDataUrl(DelegateExecution execution) {
+        String formUrl = String.valueOf(execution.getVariable(FORM_URL));
+        String requestName = String.valueOf(execution.getVariable("requestName"));
+        String requestStatus = String.valueOf(execution.getVariable("requestStatus"));
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String submittedBy = null;
+        if (authentication != null) {
```

Comment:
Can we move this block to re-usable place, as it is duplicate code ?

Reply:
The request body of this method is specific to this API having different responsibility but fetching submitted user can be moved.

Reply:
Yes, that was my comment too. Sorry for not being clear on what to change.